src/HL7/Segments/MSH.php

```diff
@@ -79,13 +79,13 @@ public function __construct(array $fields = null, array $hl7Globals = null)
      * @param int $index Index of field
      * @param string $value
      */
-    public function setField(int $index, $value = ''): bool
+    public function setField(int $index, string|int|array|null $value = ''): bool
     {
-        if (($index === 1) && strlen($value) !== 1) {
+        if (($index === 1) && (!is_string($value) || strlen($value) !== 1)) {
             return false;
         }
 
-        if (($index === 2) && strlen($value) !== 4) {
+        if (($index === 2) && (!is_string($value) || strlen($value) !== 4)) {
```

Comment:
`MSH::setTriggerEvent()` calls `$this->setField($position, $value)`, and `$value` is an array. This changes this function to take the same types as `Segment::setField()`, and ensures that index 1 or 2 must be a string. It could be better to throw an exception instead of returning false if a non-string value is given for MSH.1 or MSH.2.